### PR TITLE
Allow null values to make it to nullable database columns in provided TypeConverter impls

### PR DESCRIPTION
**Pull request by mcumings**
This change makes the TypeConverter implementations provided by the DBFlow core behave consistently with respect to how null values are represented at read-time.

### CODE REVIEW

File: flowcore/src/main/java/com/raizlabs/android/dbflow/converter/BooleanConverter.java [hunk 1]
```diff
@@ -11,6 +11,6 @@ public Integer getDBValue(Boolean model) {
 
     @Override
     public Boolean getModelValue(Integer data) {
-        return data != null && data == 1;
+        return data == null ? null : data == 1;
```
Comment (agrosner) [inline flowcore/src/main/java/com/raizlabs/android/dbflow/converter/BooleanConverter.java]: Yes we may want to return null, however this may break down conversions to ```boolean```. Can you add a test for a regular ```boolean```? thank you

File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/typeconverter/LocationConverter.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package com.raizlabs.android.dbflow.test.typeconverter;
 
 import android.location.Location;
+
```
Comment (agrosner) [inline library/src/androidTest/java/com/raizlabs/android/dbflow/test/typeconverter/LocationConverter.java]: why was this added?

**Comment by agrosner** [comment]
overall not bad and probably will make it into ```1.7.1```

File: flowcore/src/main/java/com/raizlabs/android/dbflow/converter/BooleanConverter.java (hunk 1 above)
Comment (mcumings) [inline flowcore/src/main/java/com/raizlabs/android/dbflow/converter/BooleanConverter.java]: Done
